### PR TITLE
B-20465-INT - Add Counseling Office Selection to Customer Application

### PR DESCRIPTION
**Pull request by JonSpight**
## Summary
When a customer selects an origin duty location assigned to government counseling, then they will also be asked to select a counseling office within the same GBLOC. This field will be called "Counseling Office".

### How to test as a Customer
1.  Login as a Customer
2. Select a counseling Office while creating a new move
3. Check the Moves data table for corresponding counseling_office_id
4. Create another move 
5. Select a prime counseled location (Ex. NS Mayport) and verify the customer is not prompted to pick a counseling office
7. Check the Moves data table for corresponding counseling_office_id (should be null)

### How to test as a Service Counselor
1.  Login as a Service Counselor
2. Verify the counseling queue's counseling Office column matches what was created in previous move

## 508 Screenshots
[508Screenshots-20465.pdf](https://github.com/user-attachments/files/16971315/508Screenshots-20465.pdf)

### CODE REVIEW

**Review by paulstonebraker — CHANGES_REQUESTED** [review]

File: src/components/Customer/OrdersInfoForm/OrdersInfoForm.jsx [hunk 1]
```diff
@@ -90,9 +105,28 @@ const OrdersInfoForm = ({ ordersTypeOptions, initialValues, onSubmit, onBack })
...
+                }}
                 required
               />
-
+              {dutyLocation.provides_services_counseling && (
+                <div>
+                  <Label>
+                    Select an origin duty location that most closely represents your current physical location, not
+                    where your shipment will originate, if different. This will allow a nearby transportation office to
+                    assist
```
[the remark this answers is not in the document]
Comment (JonSpight) [reply src/components/Customer/OrdersInfoForm/OrdersInfoForm.jsx:118]: good catch - fixed

File: src/pages/MyMove/AddOrders.jsx [hunk 2]
```diff
@@ -38,6 +38,9 @@ const AddOrders = ({ context, serviceMemberId, updateServiceMember, updateOrders
       origin_duty_location_id: values.origin_duty_location.id,
       spouse_has_pro_gear: false,
     };
+    if (!values.origin_duty_location.provides_services_counseling) {
+      pendingValues.counseling_office_id = undefined;
```
[the remark this answers is not in the document]
Comment (JonSpight) [reply src/pages/MyMove/AddOrders.jsx:42]: you're right, just fixed this too.

File: src/components/Table/SearchResultsTable.jsx [hunk 3]
```diff
@@ -1,26 +1,26 @@
...
-import { useTable, useFilters, usePagination, useSortBy } from 'react-table';
-import { generatePath, useNavigate } from 'react-router';
-import PropTypes from 'prop-types';
-import { Button } from '@trussworks/react-uswds';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { Button } from '@trussworks/react-uswds';
+import PropTypes from 'prop-types';
+import React, { useEffect, useMemo, useState } from 'react';
+import { generatePath, useNavigate } from 'react-router';
+import { useFilters, usePagination, useSortBy, useTable } from 'react-table';
```
[the remark this answers is not in the document]
Comment (JonSpight) [reply src/components/Table/SearchResultsTable.jsx:6]: Nope nothing was changed aside from the ordering.

**Review by paulstonebraker — APPROVED** [review]

**Comment by paulstonebraker** [comment]
Nice work!

**Comment by pambecker** [comment]
conflicts

File: src/components/Table/SearchResultsTable.jsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (pambecker) [reply src/components/Table/SearchResultsTable.jsx:6]: You didn't change anything in this file?  Then just curious how it shows up with these changes?  We don't want to introduce changes that aren't required or our backlog ususally.

[the remark this answers is not in the document]
Comment (JonSpight) [reply src/components/Table/SearchResultsTable.jsx]: there weren't any changes, the differences came from the ordering. Which changed when I was doing merge conflicts. I put them back now.

File: src/components/Customer/OrdersInfoForm/OrdersInfoForm.jsx [hunk 4]
```diff
@@ -17,12 +17,14 @@ import SectionWrapper from 'components/Customer/SectionWrapper';
...
 import { formatLabelReportByDate, dropdownInputOptions } from 'utils/formatters';
+import { showCounselingOffices } from 'services/internalApi';
 
 let originMeta;
 let newDutyMeta = '';
 const OrdersInfoForm = ({ ordersTypeOptions, initialValues, onSubmit, onBack }) => {
   const payGradeOptions = dropdownInputOptions(ORDERS_PAY_GRADE_OPTIONS);
-
+  const [dutyLocation, setDutyLocation] = useState('');
+  const [officeOptions, setOfficeOptions] = useState(null);
```
[the remark this answers is not in the document]
Comment (JonSpight) [reply src/components/Customer/OrdersInfoForm/OrdersInfoForm.jsx]: Good Call, just fixed this!

File: src/components/Customer/OrdersInfoForm/OrdersInfoForm.jsx [hunk 5]
```diff
@@ -17,12 +17,14 @@ import SectionWrapper from 'components/Customer/SectionWrapper';
...
 import { formatLabelReportByDate, dropdownInputOptions } from 'utils/formatters';
+import { showCounselingOffices } from 'services/internalApi';
 
 let originMeta;
 let newDutyMeta = '';
 const OrdersInfoForm = ({ ordersTypeOptions, initialValues, onSubmit, onBack }) => {
   const payGradeOptions = dropdownInputOptions(ORDERS_PAY_GRADE_OPTIONS);
-
+  const [dutyLocation, setDutyLocation] = useState('');
+  const [counselingOfficeOptions, setcounselingOfficeOptions] = useState(null);
```
Comment (pambecker) [inline src/components/Customer/OrdersInfoForm/OrdersInfoForm.jsx:27]: This needs to be camelCase, setCounselingOfficeOptions

Comment (JonSpight) [reply src/components/Customer/OrdersInfoForm/OrdersInfoForm.jsx:27]: good catch

**Review by paulstonebraker — APPROVED** [review]

**Review by pambecker — APPROVED** [review]

**Pull request merged**